holidays/countries/burundi.py

```diff
@@ -38,13 +37,15 @@ def __init__(self, *args, **kwargs):
         IslamicHolidays.__init__(self)
         super().__init__(*args, **kwargs)
 
-    def _add_holiday(self, *args) -> Optional[date]:
-        dt = super()._add_holiday(*args)
-        if self.observed and dt and self._is_sunday(dt):
-            super()._add_holiday(
-                "%s (Observed)" % self[dt], dt + td(days=+1)  # type: ignore[index, operator]
-            )
-        return dt
+    def _add_holiday(self, name: str, *args) -> Optional[date]:
+        dt = args if len(args) > 1 else args[0]
+        dt = dt if isinstance(dt, date) else date(self._year, *dt)
+
+        dt_added = super()._add_holiday(name, dt)
+        if self.observed and dt_added and self._is_sunday(dt_added):
+            dt_added = super()._add_holiday("%s (Observed)" % self[dt], dt + td(days=+1))
+
```

Comment:
```suggestion
        dt_added = super()._add_holiday(name, *args)
        if self.observed and dt_added and self._is_sunday(dt_added):
            dt_added = super()._add_holiday("%s (Observed)" % name, dt_added + td(days=+1))
```

Reply:
Yeah, those are for a future refactoring of `HolidayBase::_add_holiday` as it's going to stop accepting *args and will take name/dt instead. Not sure if it's better to keep it now or remove.

Reply:
Then let's keep it now.